form/events.rst

```diff
@@ -185,10 +185,6 @@ View data        Same as in ``FormEvents::POST_SET_DATA``
     See all form events at a glance in the
     :ref:`Form Events Information Table <component-form-event-table>`.
 
-.. caution::
-
-    At this point, you cannot add or remove fields to the form.
-
```

Comment:
This was already discussed in https://github.com/symfony/symfony/issues/20770
It doesn't make sense to change data on `SUBMIT` and `POST_SUBMIT`.

Reply:
This notice is not about changing the data but the form fields. Trying to do so only results in an exception being thrown on `POST_SUBMIT` event, not on `SUBMIT` event. Maybe an exception should be thrown on `SUBMIT` event too?

Reply:
Reading related discussions, it seems that `SUBMIT` event allowing to change form fields is intentional though it is suitable for a few edge cases only. What about updating the notice to say exactly that, rather than completely removing it?